src/workflows/server/server.py

```diff
@@ -855,6 +864,18 @@ async def _extract_run_params(self, request: Request, workflow: Workflow) -> tup
             context = None
             if context_data:
                 context = Context.from_dict(workflow=workflow, data=context_data)
+            elif handler_id:
```

Comment:
one thing that we might want to be cautious about is validating that its also not actually currently running to prevent duplicates writing to the same state.

Reply:
if we're to re-use handlers, I think to be more bulletproof, it still needs to check the in memory handlers. For this scenario
1. ran once, succeeded
2. started again, is in progress
3. another request to start again happens, now double writing on the handler id.

Reply:
er, I forgot it will persist before completing (facepalm)